### PR TITLE
Allow migration_lock to be specified in check_repo_status

### PR DESCRIPTION
**Pull request by greg-rychlewski**
This can be useful for example when the user wants to have advisory locks for migrations but doesn't need that level of locking for checking pending migrations.

### CODE REVIEW

**Comment by josevalim** [comment]
Woot! Nice seeing you here too @greg-rychlewski!

I think we need to bump the Ecto dependency to `~> 3.5` in the mix.exs file. Can you please do that?

**Comment by greg-rychlewski** [comment]
Oh yes sure no problem :). All done.

Is it ok that the mix.lock version is 3.9 or should that be 3.5?

**Comment by josevalim** [comment]
It is fine to be 3.9 I think!

**Comment by josevalim** [comment]
It seems we will also have to bump CI and mix.exs to use at least Elixir v1.10. :)

File: lib/phoenix_ecto/check_repo_status.ex [hunk 1]
```diff
@@ -9,12 +9,15 @@ defmodule Phoenix.Ecto.CheckRepoStatus do
...
     * `:otp_app` - name of the application which the repos are fetched from
     * `:migration_paths` - a function that accepts a repo and returns a migration directory, or a list of migration directories, that is used to check for pending migrations
+    * `:migration_lock` - the locking strategy used by the Ecto Adapter when checking for pending migrations. Set to `false` to disable migration locks.
   """
 
   @behaviour Plug
 
   alias Plug.Conn
 
+  @migration_opts [:migration_lock]
```
Comment (josevalim) [inline lib/phoenix_ecto/check_repo_status.ex:19]:
```suggestion
  @migration_opts [:migration_lock]
  @compile {:no_warn_undefined, {Ecto.Migrator, :migrations, 3}}
```

**Comment by greg-rychlewski** [comment]
Updated the CI to  Elixir 1.10/OTP 21 and Elixir 1.14/OTP 25

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by dbernheisel** [comment]
you beat me to it :)

**Pull request merged**